enhancements/update/update-status-api.md

```diff
@@ -0,0 +1,410 @@
+
+We do not expect users to interact with `UpdateStatus` resources directly; the API is intended to be used mainly by tooling. A typical `UpdateStatus` instance is likely to be quite verbose.
+
+#### Update Insights
+
+Update Insights are units of update status/health information. There are two types of update insights: status insights and health insights.
+
+##### Update Status Insights
+ 
+Update Status Insights are directly tied to the update process, regardless of whether it is proceeding smoothly or not. Status Insights expose the status of a single resource that is directly involved in the update process, usually a resource that:
+
+1. Either has a notion of "being updated," such as a `Node` or `ClusterOperator`
+1. or represents a higher-level abstraction, such as a `ClusterVersion` resource (represents the control plane) or `MachineConfigPool` (represents a pool of nodes).
+
+Initially, the USC will produce status insights for `ClusterVersion`, `ClusterOperator`, `MachineConfigPool`, and `Node` resources.
+
+##### Update Health Insights
+
+Update Health Insights report a state or condition in the cluster that is abnormal or negative and either affects or is affected by the update. Ideally, none would be generated in a standard healthy update. Health insights should communicate a condition that warrants attention by the cluster administrator and should be actionable. Links to resources helpful in the given situation should accompany health insights. Health insights can reference multiple resources involved in the situation. For example, a health insight exposing a failure to drain a node would reference the `Node`, the `Pod` that fails to be drained, and the `PodDisruptionBudget` that protects it. Individual resources can be involved in multiple insights.
+
+#### Update Status Controller
+
+The Update Status Controller (USC) is a new component in OpenShift that will be deployed directly by the CVO and is being treated as its operand. This is uncommon in OpenShift (listed as a [drawback][this-drawbacks]), as the CVO typically deploys only second-level operators as its operands. In this model, the USC (providing cluster functionality) would normally be an operand, and we would need to invent a new cluster operator to manage it. Because the CVO is directly involved in updates, such an additional layer does not have value, at least for now. If managing the USC shows to be a sufficiently complex problem in practice, we can consider introducing a new operator to manage it.
+
+The Update Status Controller will run a primary controller that will maintain the `UpdateStatus` resource. The resource has no `.spec`, so the controller will ensure the resource exists and its `.status` content is up-to-date and correct. It will determine the `status` subresource content by receiving and processing insights from the other controllers in the USC.
```

Comment:
> receiving and processing insights from the other controllers in the USC

Is there a message bus involved? How does it receive insights from other places?

Reply:
It is not proposed as a full distributed system yet, so I think of inter-controller (within a single process) communication as an implementation detail. I have it partially working and I owe even my team a high-level overview of the algorithm so I will write it and link it from here (but I'd like to avoid including it in the enhancement because I think it is a detail for now). 

High-level: multiple informers send messages through a Go channel to a consumer. No bi-directional communication. Bit of protocol there to handle outdated state etc.

Reply:
In the future, when this does become a distributed system, have you thought about what that message bus would look like? It feels like an important step to understand this if we are considering the current proposal a temporary route